discord/template.py

```diff
@@ -150,18 +154,18 @@ def _store(self, data: TemplatePayload):
             source_serialised = data['serialized_source_guild']
             source_serialised['id'] = id
             state = _PartialTemplateState(state=self._state)
-            guild = Guild(data=source_serialised, state=state)  # type: ignore
+            guild = Guild(data=source_serialised, state=state)  # type: ignore - Guild expects a ConnectionState, we're passing a _PartialTemplateState
```

Comment:
Should `Guild` be updated to take a protocol that both `ConnectionState` and `_PartialTemplateState` meet?

Reply:
That's an open question worth considering down the line but a lot of models take in fake connection states and it's more trouble than it's worth compared to just lying like is done here.